src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -673,22 +821,229 @@ public override void VisitTagHelperBlock(TagHelperChunkGenerator chunkGenerator,
+                {
+                    foreach (var associatedDescriptor in associatedDescriptors)
+                    {
+                        var associatedAttributeDescriptor = associatedDescriptor.BoundAttributes.First(a =>
+                        {
+                            return TagHelperMatchingConventions.CanSatisfyBoundAttribute(attributeName, a);
+                        });
+
+                        var setTagHelperProperty = new TagHelperPropertyIntermediateNode()
+                        {
+                            AttributeName = attributeName,
+                            BoundAttribute = associatedAttributeDescriptor,
+                            TagHelper = associatedDescriptor,
+                            AttributeStructure = node.TagHelperAttributeInfo.AttributeStructure,
+                            Source = BuildSourceSpanFromNode(attributeValueNode),
+                            IsIndexerNameMatch = TagHelperMatchingConventions.SatisfiesBoundAttributeIndexer(attributeName, associatedAttributeDescriptor),
+                        };
+
+                        _builder.Push(setTagHelperProperty);
+                        VisitAttributeValue(attributeValueNode);
+                        _builder.Pop();
+                    }
+                }
+                else
+                {
+                    var addHtmlAttribute = new TagHelperHtmlAttributeIntermediateNode()
+                    {
+                        AttributeName = attributeName,
+                        AttributeStructure = node.TagHelperAttributeInfo.AttributeStructure
+                    };
+
+                    _builder.Push(addHtmlAttribute);
+                    VisitAttributeValue(attributeValueNode);
+                    _builder.Pop();
+                }
+
+                return node;
+            }
+
+            private void VisitAttributeValue(SyntaxNode node)
+            {
+                if (node == null)
+                {
+                    return;
+                }
+
+
+                IReadOnlyList<SyntaxNode> children = node.ChildNodes();
+                var position = node.Position;
+                if (children.First() is MarkupBlockSyntax markupBlock &&
+                    markupBlock.Children.Count == 2 &&
+                    markupBlock.Children[0] is MarkupTextLiteralSyntax &&
+                    markupBlock.Children[1] is MarkupEphemeralTextLiteralSyntax)
+                {
+                    // This is a special case when we have an attribute like attr="@@foo".
+                    // In this case, we want the foo to be written out as HtmlContent and not HtmlAttributeValue.
+                    Visit(markupBlock);
+                    children = children.Skip(1).ToList();
+                    position = children.Count > 0 ? children[0].Position : position;
+                }
+
+                if (children.All(c => c is MarkupLiteralAttributeValueSyntax))
```

Comment:
Shouldn't this be elseif? Maybe i'm misunderstanding but doesn't the first bit result in everything else returning false?